internal/resources/kubernetessecret/resource_secret.go

```diff
@@ -105,56 +110,140 @@ func getSecretSchema(isDataSource bool) map[string]*schema.Schema {
+		return diag.Errorf("Unable to create Tanzu Mission Control secret entry; Scope full name is empty")
 	}
 
-	d.SetId(secretResponse.Secret.Meta.UID)
+	var (
+		UID  string
+		meta = common.ConstructMeta(d)
+	)
 
-	if d.Get(ExportKey).(bool) {
-		secretexportRequest := &secretexportmodel.VmwareTanzuManageV1alpha1ClusterNamespaceSecretExportRequest{
-			SecretExport: &secretexportmodel.VmwareTanzuManageV1alpha1ClusterNamespaceSecretExport{
-				FullName: constructFullnameSecetExport(d),
-				Meta:     common.ConstructMeta(d),
-			},
+	switch scopedFullnameData.Scope {
+	case commonscope.ClusterScope:
+		if scopedFullnameData.FullnameCluster != nil {
+			secretReq := &clustersecretmodel.VmwareTanzuManageV1alpha1ClusterNamespaceSecretRequest{
+				Secret: &clustersecretmodel.VmwareTanzuManageV1alpha1ClusterNamespaceSecret{
+					FullName: scopedFullnameData.FullnameCluster,
+					Meta:     meta,
+					Spec:     spec.ConstructSpecForClusterScope(d),
+				},
+			}
+
+			secretResponse, err := config.TMCConnection.SecretResourceService.SecretResourceServiceCreate(secretReq)
+			if err != nil {
+				return diag.FromErr(errors.Wrapf(err, "Unable to create Tanzu Mission Control cluster secret entry, name : %s", secretName))
+			}
+
+			UID = secretResponse.Secret.Meta.UID
+		}
+	case commonscope.ClusterGroupScope:
+		if scopedFullnameData.FullnameClusterGroup != nil {
+			secretReq := &clustergroupsecretmodel.VmwareTanzuManageV1alpha1ClustergroupNamespaceSecretRequest{
+				Secret: &clustergroupsecretmodel.VmwareTanzuManageV1alpha1ClustergroupNamespaceSecretSecret{
+					FullName: scopedFullnameData.FullnameClusterGroup,
+					Meta:     meta,
+					Spec:     spec.ConstructSpecForClusterGroupScope(d),
+				},
+			}
+
+			secretResponse, err := config.TMCConnection.ClusterGroupSecretResourceService.SecretResourceServiceCreate(secretReq)
+			if err != nil {
+				return diag.FromErr(errors.Wrapf(err, "Unable to create Tanzu Mission Control cluster group secret entry, name : %s", secretName))
+			}
+
+			UID = secretResponse.Secret.Meta.UID
 		}
+	case commonscope.UnknownScope:
+		return diag.Errorf("no valid scope type block found: minimum one valid scope type block is required among: %v. Please check the schema.", strings.Join(scope.ScopesAllowed[:], `, `))
+	}
 
-		_, err = config.TMCConnection.SecretExportResourceService.SecretExportResourceServiceCreate(secretexportRequest)
+	d.SetId(UID)
 
+	if d.Get(ExportKey).(bool) {
+		err := createDeleteSecretExport(true, scopedFullnameData, config, d)
 		if err != nil {
-			return diag.FromErr(errors.Wrapf(err, "unable to create Tanzu Mission Control secret export entry, name : %s", NameKey))
+			return diag.FromErr(err)
 		}
 	}
 
 	return dataSourceSecretRead(ctx, d, m)
 }
 
-func resourceSecretDelete(_ context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+func resourceSecretDelete(_ context.Context, d *schema.ResourceData, m interface{}) (diags diag.Diagnostics) {
```

Comment:
Where are we setting ID to nil in the delete call?

Reply:
at line 249 we are calling `schema.RemoveFromState(d, m)` in delete method which make id to nil.